clang/lib/CodeGen/CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
+if(EXISTS ${LLVM_MAIN_SRC_DIR}/include/llvm/CodeGen AND CLANG_BUILT_STANDALONE)
+  add_subdirectory(${LLVM_MAIN_SRC_DIR}/include/llvm/CodeGen llvm/lib/CodeGen)
+endif()
```

Comment:
IIRC, even in the standalone case, you need to identify the LLVM build and pass that in via `LLVM_DIR`, why is that not sufficient for finding this include path?

Reply:
I'm not sure, are you meaning to use `LLVM_DIR` instead of `LLVM_MAIN_SRC_DIR`?

Reply:
No, `LLVM_DIR` is passed to the CMake configure step to locate the build of LLVM. That should wire up the include path that should be needed either through a dependency on LLVMCodeGen or through a global include search path.

Reply:
I'm not exactly sure what you're meaning.

Reply:
The issue is that the `vt_gen` target is undefined. This is _expected_. The target should be defined by LLVM, which is a dependency. You should be wiring up the targets from LLVM into the clang build, which would provide the dependency and avoid this issue. The target definition is setup by `find_package(LLVM)` which is why you need to specify `-D LLVM_DIR=...` to the CMake invocation when configuring clang standalone. The normal build is unified and the target is defined when clang is configured.

Reply:
Ok, so the solution is to set `-DLLVM_DIR` which is done by `find_package`? We already have LLVM inside Clang's CMake search path so I am not sure why the error in the PR toplevel comment is happening.

Reply:
No, passing `-D LLVM_DIR=...` is done by the user to satisfy the `find_package`. If you do not pass it the build tree, it will attempt to find an installed copy of LLVM which may not be defining the target.

Reply:
https://github.com/llvm/llvm-project/blob/d61b2590f8e360695a5298311855c8649337969f/clang/CMakeLists.txt#L40-L41

Looks more like `find_package` is using `HINTS` to find where LLVM is at and it's required so it would be resolved and throwing an error earlier if it wasn't satisfied.

Reply:
Also, I just tried with `LLVM_DIR=${llvm}/lib/cmake/llvm` and got the same error as before.

Reply:
Was that on a clean configure + build?

Reply:
Yes, Nix always performs a clean configure and build.

Reply:
https://github.com/llvm/llvm-project/blob/main/llvm/cmake/modules/CMakeLists.txt#L28-L38 I suspect that you just need to extend that with `vt_gen`.

Reply:
Looks like CMake accepted that change, let's see if it actually compiles.

Reply:
Sweet, that worked.